src/test/java/io/cdap/plugin/ariba/source/metadata/AribaSchemaGeneratorTest.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.ariba.source.metadata;
+
+import io.cdap.plugin.ariba.source.util.ResourceConstants;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class AribaSchemaGeneratorTest {
+
+  @Test
+  public void testAribaSchemaGenerator() {
+    AribaColumnMetadata.Builder columnDetail = AribaColumnMetadata.builder();
+    columnDetail.viewTemplateName("name2")
+      .name("name").isPrimaryKey(false).type(ResourceConstants.OBJECT).size(0)
+      .isCustomField(false).scale(0).precision(0)
+      .childList(null);
+    AribaColumnMetadata columnList = columnDetail.build();
+    AribaSchemaGenerator aribaSchemaGenerator = new AribaSchemaGenerator(Collections.singletonList(columnList));
+    Assert.assertEquals("RECORD", aribaSchemaGenerator.buildSchema().getType().toString());

```

Comment:
can we validate the contents of the schema?

Reply:
Fixed